src/sqlfluff/dialects/dialect_databricks.py

```diff
@@ -377,3 +378,58 @@ class AliasExpressionSegment(sparksql.AliasExpressionSegment):
             ),
         ),
     )
+
+
+class GeneratedColumnDefinitionSegment(sparksql.GeneratedColumnDefinitionSegment):
+    """A generated column definition, e.g. for CREATE TABLE or ALTER TABLE.
+
+    https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html
+    """
+
+    match_grammar: Matchable = Sequence(
+        Ref("SingleIdentifierGrammar"),  # Column name
+        Ref("DatatypeSegment"),  # Column type
+        Bracketed(Anything(), optional=True),  # For types like DECIMAL(3, 2)
+        OneOf(
+            Sequence(
+                "GENERATED",
+                "ALWAYS",
+                "AS",
+                Bracketed(
+                    OneOf(
+                        Ref("FunctionSegment"),
+                        Ref("BareFunctionSegment"),
+                    ),
+                ),
+            ),
+            Sequence(
+                "GENERATED",
+                OneOf(
+                    "ALWAYS",
+                    Sequence("BY", "DEFAULT"),
+                ),
+                "AS",
+                "IDENTITY",
+                Bracketed(
+                    Sequence(
+                        Sequence(
```

Comment:
is it possible these two can be the other way round?

Reply:
I didn't think about that, but I just checked, and it seems "STARTS WITH" does have to come before "INCREMENT BY".